habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -469,6 +469,178 @@ def step(self, *args, is_last_action, **kwargs):
+        super().__init__(*args, config=config, sim=sim, **kwargs)
+        self._lin_collision_threshold = self._config.lin_collision_threshold
+        self._ang_collision_threshold = self._config.ang_collision_threshold
+        self._x_offset = self._config.x_offset
+        self._y_offset = self._config.y_offset
+        self._enable_lateral_move = self._config.enable_lateral_move
+        assert len(self._x_offset) == len(self._y_offset)
+
+    @property
+    def action_space(self):
+        lim = 20
+        if self._enable_lateral_move:
+            return spaces.Dict(
+                {
+                    self._action_arg_prefix
+                    + "base_vel": spaces.Box(
+                        shape=(3,), low=-lim, high=lim, dtype=np.float32
+                    )
+                }
+            )
+        else:
+            return spaces.Dict(
+                {
+                    self._action_arg_prefix
+                    + "base_vel": spaces.Box(
+                        shape=(2,), low=-lim, high=lim, dtype=np.float32
+                    )
+                }
+            )
+
+    def collision_check(
+        self, trans, target_trans, center_end_pos, target_rigid_state
+    ):
+        """
+        trans: the transformation of the current location of the robot
+        target_trans: the transformation of the target location of the robot given the center original Navmesh
+        center_end_pos: the original center end position of the robot
+        target_rigid_state: the target state of the robot given the center original Navmesh
+        """
+        num_check_cylinder = len(self._x_offset)
+        # Get the offset positions
+        cur_pos = []
+        goal_pos = []
+        for i in range(num_check_cylinder):
+            cur_pos.append(
+                trans.transform_point(
+                    np.array([self._x_offset[i], 0, self._y_offset[i]])
+                )
+            )
+            goal_pos.append(
+                target_trans.transform_point(
+                    np.array([self._x_offset[i], 0, self._y_offset[i]])
+                )
+            )
+
+        # For step filter of offset positions
+        end_pos = []
+        for i in range(num_check_cylinder):
+            pos = self._sim.step_filter(cur_pos[i], goal_pos[i])
+            # Sanitize the height
+            pos[1] = cur_pos[i][1]
+            end_pos.append(pos)
+
+        # For step filter of the center original pos
+        center_end_pos[1] = trans.translation[1]
+
+        # Planar move distance clamped by NavMesh
+        move = []
+        for i in range(num_check_cylinder):
+            move.append((end_pos[i] - goal_pos[i]).length())
+        center_move = (center_end_pos - trans.translation).length()
+
+        # Wrap the move direction if we use sliding
+        # Find the largest diff moving direction, which means that there is a collision in that cylinder
+        max_idx = np.argmax(move + [center_move])
```

Comment:
Yes, it will be applied no matter what the value of `if_sliding ` is. If `if_sliding` is true, then we will slide the robot, based on the computation from this function `collision_check `